openupgrade_scripts/scripts/account/14.0.1.1/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,648 @@
+---Models in module 'account'---
+obsolete model account.accrual.accounting.wizard [transient]
+new model account.automatic.entry.wizard [transient]
+# NOTHING TO DO: Accrual Entries and Transfer Entries merged into one wizard
+
+obsolete model account.fiscal.year
+obsolete model account.reconciliation.widget [abstract]
+obsolete model report.account.report_agedpartnerbalance [abstract]
+# NOTHING TO DO: moved to enterprise
+
+new model account.reconcile.model.line
+new model account.reconcile.model.line.template
+# DONE: post-migration: create line records from parent model
+
+new model account.tax.report
+# DONE: post-migration: create parent records from line model
+
+new model account.group.template
+new model account.reconcile.model.partner.mapping
+# NOTHING TO DO: new models
+
+new model account.resequence.wizard [transient]
+new model account.tour.upload.bill [transient]
+new model account.tour.upload.bill.email.confirm [transient]
+new model report.account.report_invoice [abstract]
+new model sequence.mixin [abstract]
+# NOTHING TO DO: abstract and transient models
+
+---Fields in module 'account'---
+account      / account.account          / _order                        : _order is now 'is_off_balance, code, company_id' ('code')
+account      / account.account          / is_off_balance (boolean)      : NEW hasdefault, isfunction: function, stored
+# NOTHING TO DO: new field added to filter accounts.
+
+account      / account.account          / allowed_journal_ids (many2many): NEW relation: account.journal
+# NOTHING TO DO: New feature. In v13 you selected the allowed accounts form the journal form, now you can select the allowed journals from the account form
```

Comment:
```suggestion
# DONE: renamed many2many table. In v13 you selected the allowed accounts from the journal form, now you can select the allowed journals from the account form
```

Reply:
oh, not really, don't be confused :S

In `account.journal` you still have `account_control_ids = fields.Many2many('account.account', 'journal_account_control_rel', 'journal_id', 'account_id', string='Allowed accounts'`, which this field already existed in v13 but the table changed and was renamed.

And then in `account.account` you have `allowed_journal_ids = fields.Many2many('account.journal', string="Allowed Journals"` which uses `account_account_account_journal_rel` table which is a true new feature if you analyze the odoo code.

account_account_account_journal_rel != journal_account_control_rel

Reply:
I clarified the comment in the txt.

Reply:
Oh man. I can't believe they did this. Thanks for clarifying!

Reply:
Then again, I guess it makes sense because it can be the case that you want to restrict an account to only be used in a specific journal but at the same time do not want to restrict that journal for all other accounts.